### PR TITLE
Password storing methods

### PR DESCRIPTION
**Pull request by Zarathustra4**

### CODE REVIEW

File: src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java [hunk 1]
```diff
@@ -17,7 +36,56 @@ public class PasswordController {
...
+    public PasswordResponseDto generatePassword(){
         return passwordEntityService.generatePassword();
     }
+
+    @PostMapping("/generate")
+    public PasswordResponseDto generateAndStorePassword(
+            @AuthenticationPrincipal AuthUserDetailsDto userDetailsDto,
+            @RequestBody PasswordRequestDto passwordRequestDto)
+
+            throws NameNotFoundException, NoSuchPaddingException, IllegalBlockSizeException,
```
Comment (justme1135) [inline src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java]: Too many exceptions
Maybe you should catch them on service level and re-throw some custom exception

File: src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java [hunk 2]
```diff
@@ -17,7 +36,56 @@ public class PasswordController {
...
+            NoSuchAlgorithmException, BadPaddingException, InvalidKeyException
+
+    {
+        return passwordEntityService.generateAndStorePassword(passwordRequestDto, userDetailsDto);
+    }
+
+    @PostMapping
+    public ResponseEntity<String> storePassword(@AuthenticationPrincipal AuthUserDetailsDto userDetailsDto,
+                                                        @RequestBody PasswordRequestDto passwordRequest)
+            throws NameNotFoundException, NoSuchPaddingException, IllegalBlockSizeException,
```
Comment (justme1135) [inline src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java]: +1

File: src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java [hunk 3]
```diff
@@ -17,7 +36,56 @@ public class PasswordController {
...
+            NoSuchAlgorithmException, BadPaddingException, InvalidKeyException {
+
+        passwordEntityService.savePassword(passwordRequest, userDetailsDto);
+        return new ResponseEntity<>("Password was stored", HttpStatus.OK);
+    }
+
+    @GetMapping("/{serviceId}")
+    public PasswordResponseDto getPassword(@AuthenticationPrincipal AuthUserDetailsDto userDetailsDto, @PathVariable Long serviceId)
```
Comment (justme1135) [inline src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java]: +1

File: src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java [hunk 4]
```diff
@@ -17,7 +36,56 @@ public class PasswordController {
...
+            throws NoSuchPaddingException, IllegalBlockSizeException, NoSuchAlgorithmException,
+            BadPaddingException, InvalidKeyException {
+
+        return passwordEntityService.getPassword(serviceId, userDetailsDto);
+    }
+
+    @PutMapping
+    public ResponseEntity<String> changePassword(
+            @AuthenticationPrincipal AuthUserDetailsDto userDetailsDto,
```
Comment (justme1135) [inline src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java:65]: +1

File: src/main/java/com/passpnu/passwordmanager/entity/PasswordEntity.java [hunk 5]
```diff
@@ -26,16 +27,21 @@ public class PasswordEntity {
...
     private Long id;
 
-    @NotBlank
     @ManyToOne
-    @JoinColumn(name="service_id", nullable = false)
+    @JoinColumn(name="service_id", nullable = false, updatable = false, insertable = false)
     private ServiceEntity service;
 
+    @Column(name="service_id")
+    private Long serviceId;
```
Comment (justme1135) [inline src/main/java/com/passpnu/passwordmanager/entity/PasswordEntity.java:31]: I am not 100% sure whether you need both `service` and `serviceId` when you can easily get one from the other

File: src/main/java/com/passpnu/passwordmanager/entity/PasswordEntity.java [hunk 6]
```diff
@@ -26,16 +27,21 @@ public class PasswordEntity {
...
     @NotBlank
     private String password;
 
     @ManyToOne
-    @JoinColumn(name="user_id", nullable = false)
+    @JoinColumn(name="user_id", nullable = false, insertable = false, updatable = false)
     private UserEntity user;
 
+    @Column(name="user_id")
+    private Long userId;
```
Comment (justme1135) [inline src/main/java/com/passpnu/passwordmanager/entity/PasswordEntity.java:38]: +1

File: src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java (hunk 1 above)
Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java]: fixed

File: src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java (hunk 2 above)
Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java]: fixed

File: src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java (hunk 3 above)
Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java]: fixed

File: src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java (hunk 4 above)
Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/controller/PasswordController.java:65]: fixed

File: src/main/java/com/passpnu/passwordmanager/entity/PasswordEntity.java (hunk 5 above)
Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/entity/PasswordEntity.java:31]: fixed

File: src/main/java/com/passpnu/passwordmanager/entity/PasswordEntity.java (hunk 6 above)
Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/entity/PasswordEntity.java:38]: fixed

**Review by justme1135 — APPROVED** [review]

**Pull request merged**